src/sagas/index.js

```diff
@@ -143,85 +143,115 @@ export function* transformAndPermitVm (vm) {
  */
 export function* fetchByPage () {
   const [
-    vmsPage,
-    vmsExpectMorePages,
-    poolsPage,
-    poolsExpectMorePages,
+    vmsCount, vmsPage, vmsExpectMorePages,
```

Comment:
if `vmsExpectMorePages` is false we should trigger early return and avoid all the conditional logic below (the same applies to to pools)

Reply:
Most of it will be tossed to the reducers in that bit of refactor-to-a-single-put work anyway.  It's easier to just grab and put both VMs and Pools....